### PR TITLE
test: api-tests from js to typescript

### PR DESCRIPTION
**Pull request by tormoseng**
Re-write the API-tests in TypeScript. Summary:
- The tests are in `test/src`
- Webpack packs runnable k6 scripts into `test/dist`
- Response types are used throughout the tests, except for some v1 endpoints that are only used by older app versions

### CODE REVIEW

**Comment by tormoseng** [comment]
The tests are successful at leats: https://github.com/AtB-AS/atb-bff/actions/runs/3471940040

**Comment by tormoseng** [comment]
Tests run OK: https://github.com/AtB-AS/atb-bff/actions/runs/3513426396

File: .github/workflows/api-tests-ts.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+jobs:
+  api-test:
+    name: API-tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: tor/api-tests-typescript
+
```
Comment (rosvik) [inline .github/workflows/api-tests-ts.yml:18]: Should the branches be updated here before merging to main?

File: test/src/v1/types/geocoder.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  };
+  properties: {
+    id: string;
+    gid: string;
+    layer: string;
+    source: string;
+    source_id: string;
+    name: string;
+    street: string;
+    distance: 1.895;
```
Comment (rosvik) [inline test/src/v1/types/geocoder.ts:15]: This looks weirdly specific for a type 🤔 `number` instead?

File: test/src/v1/types/geocoder.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  properties: {
+    id: string;
+    gid: string;
+    layer: string;
+    source: string;
+    source_id: string;
+    name: string;
+    street: string;
+    confidence: string;
+    distance: 1.895;
```
Comment (rosvik) [inline test/src/v1/types/geocoder.ts:42]: `number`?

File: test/src/v1/types/journey.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    toPlace: {
+      name: string;
+    };
+    serviceJourney: {
+      id: string;
+    };
+  }>;
+}>;
+
+export type TripsSingleSimplifiedResponseType = TripsSimplifiedResponseType[0];
```
Comment (rosvik) [inline test/src/v1/types/journey.ts:18]: Would it not make more sense to make the `TripsSingleSimplifiedResponseType` be the object, then `TripsSimplifiedResponseType = TripsSingleSimplifiedResponseType[]`?

File: test/src/v1/types/geocoder.ts (hunk 2 above)
Comment (tormoseng) [reply test/src/v1/types/geocoder.ts]: Indeed!

File: test/src/v1/types/journey.ts (hunk 4 above)
Comment (tormoseng) [reply test/src/v1/types/journey.ts]: Nice catch! I'll change.

**Comment by tormoseng** [comment]
All tests OK on `main`: https://github.com/AtB-AS/atb-bff/actions/runs/3532417591

**Pull request merged**